devimint/src/federation.rs

```diff
@@ -371,9 +371,14 @@ impl Federation {
         let finality_delay = self.get_finality_delay().await?;
         let block_count = self.bitcoind.get_block_count().await?;
         let expected = block_count.saturating_sub(finality_delay.into());
-        cmd!(self.client, "dev", "wait-block-count", expected)
-            .run()
-            .await?;
+        poll("await-block-sync", || async {
```

Comment:
This looks OK, but would great to know what exactly is failing there. IIRC `stderr` of commands ran by devimint is printed out.

Possibly we should make the `fedimint-cli dev wait-block-count` retry etc. as it is used in other places, etc. We might be assuming it can just keep waiting.

Reply:
https://github.com/fedimint/fedimint/pull/4800

Reply:
I tried that branch out locally, but my computer still has the same issue :(

Reply:
This appears to be where the error is being thrown:

https://github.com/fedimint/fedimint/blob/747392ecd73bcf5304a19c2cb034911345ebece8/fedimint-client/src/lib.rs#L804-L825

`client_secret` is `None`, so `load_decodable_client_secret_opt()` returns `Ok(None)` which causes `load_decodable_client_secret()` to bail.

Reply:
@maan2003 That 60s timeout is too large to make a difference in practice.

Oh, I see it now

```
            Command::Dev(DevCmd::WaitBlockCount { count: target }) => {
                task::timeout(Duration::from_secs(60), async move {
                    let client = self.client_open(&cli).await?;
                    loop {
                        let wallet = client.get_first_module::<WalletClientModule>();
```

That `client_open` should be *inside* the loop. And also - all errors should lead to retry, not finishing early. We basically have no retry on error there, only retry on success.

Reply:
Let's discuss in #4800 and depending what are going to do, we should land or close this one.

@tvolk131 This is a great find one way or the other, and addressing it will improve robustness of all our scripts, so :+1: even if we don't land this PR.